Desktop_Interface/uartstyledecoder.cpp

```diff
@@ -2,41 +2,43 @@
 #include <QDebug>
 #include <cassert>
 
-uartStyleDecoder::uartStyleDecoder(QObject *parent_in) : QObject(parent_in)
+uartStyleDecoder::uartStyleDecoder(QObject *parent_in)
+	: QObject(parent_in)
+	, parent{static_cast<isoBuffer*>(parent_in)}
+	, m_serialBuffer{SERIAL_BUFFER_LENGTH}
 {
-    parent = (isoBuffer *) parent_in;
 
 	// Begin decoding SAMPLE_DELAY seconds in the past.
 	serialPtr_bit = (int)(parent->m_back * 8 - SERIAL_DELAY * parent->m_sampleRate_bit + parent->m_bufferLen * 8) % (parent->m_bufferLen*8);
 
-    updateTimer = new QTimer();
-    updateTimer->setTimerType(Qt::PreciseTimer);
-    updateTimer->start(CONSOLE_UPDATE_TIMER_PERIOD);
-    connect(updateTimer, SIGNAL(timeout()), this, SLOT(updateConsole()));
+    m_updateTimer.setTimerType(Qt::PreciseTimer);
+    m_updateTimer.start(CONSOLE_UPDATE_TIMER_PERIOD);
+    connect(&m_updateTimer, &QTimer::timeout, this, &uartStyleDecoder::updateConsole);
 
-    serialBuffer = new isoBufferBuffer(SERIAL_BUFFER_LENGTH);
-
-    if(parent->m_channel == 1) console = parent->m_console1;
-    else if(parent->m_channel == 2) console = parent->m_console2;
-    else qFatal("Nonexistant console requested in uartStyleDecoder::serialDecode");
+    if (parent->m_channel == 1)
```

Comment:
This would need to be a whole other PR, but rather than constantly reading the internal state of the parent, many of these variables should be passed in as constructor arguments.  The one or two that need to be polled should be functions in isoBuffer.